src/postgres/tree/PostgresServerTreeItem.ts

```diff
@@ -68,15 +75,30 @@ export class PostgresServerTreeItem extends AzureParentTreeItem<ISubscriptionCon
     }
 
     public async loadMoreChildrenImpl(_clearCache: boolean): Promise<AzExtTreeItem[]> {
-        const client: PostgreSQLManagementClient = createAzureClient(this.root, PostgreSQLManagementClient);
-        const listOfDatabases: DatabaseListResult = await client.databases.listByServer(this.resourceGroup, this.name);
+        if (!(this.connectionString.username && this.connectionString.password)) {
+            const { username, password } = await this.getCredentials();
+            if (username && password) {
+                await this.updateConnectionString(username, password);
```

Comment:
That's true, this will cause an error before allowing the user to enter the credentials for Azure scenario. 
I like the first option better for now, given there is no credentials support for attached scenario. It is large piece of code to move to the ServerTreeItem, I would hold off on that for now.